improver/metadata/utilities.py

```diff
@@ -111,6 +111,9 @@ def generate_mandatory_attributes(diagnostic_cubes):
     Args:
         diagnostic_cubes (list):
```

Comment:
You are right.  There's a reason I did it like this, based on the understanding (possibly misunderstanding) that there's no extra cost to passing larger objects in python, since they're effectively passed by reference.  Might be worth discussing offline.